store/views.py

```diff
@@ -53,3 +57,96 @@ def get(self, request):
             })
 
         return JsonResponse({'result': result}, status=200)
+
+class ProductDetailsView(View):
+    def get(self,request,product_id):
+        result = []
+        try:
+            product    = Product.objects.filter(id=product_id)
+
+            if not product:
+                return JsonResponse({"message":"NOT_EXIST"}, status=400)
+
+            product_info   = product.get()
+            result.append({
+                'id'           : product_info.id,
+                'title'        : product_info.title,
+                'market_price' : product_info.market_price,
+                'sale_price'   : product_info.sale_price,
+                'description'  : product_info.description,
+                'product_img'  : product_info.thumnail_url,   
+            })
+            return JsonResponse({'result': result}, status=200)
+        
+        except KeyError:
+            return JsonResponse({"message":"KEYERROR"}, status=400)
+
+class Reviews(View):
+    @login_check
+    def post(self,request,product_id):
+        try:
+            data = json.loads(request.body)
+            user = request.user
+            Review.objects.create(
+                content          = data['content'],
+                rate             = data['rate'],
+                review_image_url = data['review_image_url'],
+                user             = user,
+                product_id       = product_id
+            )
+            return JsonResponse({'message' : 'SUCCESS'}, status = 200)
+
+        except KeyError as ex:
+            return JsonResponse({'message' : 'KEY_ERROR_' + ex.args[0]}, status = 400)
+        except Exception as ex:
+            return JsonResponse({'message' : 'ERROR_' + ex.args[0]}, status = 400)
+    
+    @login_check
+    def get(self,request,product_id):
+        try:
+            page    = int(request.GET.get('page', 1))
+            product = Product.objects.get(id = product_id)
+            reviews_list = Review.objects.order_by('-id').select_related('product','user').filter(product_id=product.id)
```

Comment:
```suggestion
            reviews_list = Review.objects.order_by('-id').select_related('product','user').filter(product_id=product_id)
```
108번 줄에서 선언하신 `product`를 `id`를 뽑아내실 때에만 사용하고 있으시네요 :) 이미 우리는 product_id를 알고 있기 때문에 user 객체를 꼭 가져와서 id를 꺼내시지는 않으셔도 됩니다 :)

Reply:
아.... 넵!